Xamarin.Essentials/Permissions/Permissions.shared.cs

```diff
@@ -1,27 +1,131 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Reflection;
 using System.Threading.Tasks;
 
 namespace Xamarin.Essentials
 {
-    internal static partial class Permissions
+    public static partial class Permissions
     {
-        internal static void EnsureDeclared(PermissionType permission) =>
-            PlatformEnsureDeclared(permission, true);
+        public static Task<PermissionStatus> CheckStatusAsync<TPermission>()
+            where TPermission : BasePermission, new() =>
+                new TPermission().CheckStatusAsync();
 
-        internal static bool EnsureDeclared(PermissionType permission, bool throwIfMissing) =>
-            PlatformEnsureDeclared(permission, throwIfMissing);
+        public static Task<PermissionStatus> RequestAsync<TPermission>()
+            where TPermission : BasePermission, new() =>
+                new TPermission().RequestAsync();
 
-        internal static Task<PermissionStatus> CheckStatusAsync(PermissionType permission) =>
-            PlatformCheckStatusAsync(permission);
+        internal static void EnsureDeclared<TPermission>()
+            where TPermission : BasePermission, new() =>
+                new TPermission().EnsureDeclared();
 
-        internal static Task<PermissionStatus> RequestAsync(PermissionType permission) =>
-            PlatformRequestAsync(permission);
+        public static void ShowSettingsUI() =>
```

Comment:
Do you think that this is confusing that it may not open the permissions screen? I think it is possible to do on some platforms.